### PR TITLE
[M-0008] extend typeORM Repository with PetRepository

### PR DESCRIPTION
**Pull request by bigLucas**
### Reviewer:
@Aristevao 
### Description:
- Add the extend of the typeORM Repository on PetRepository, and;
- Update the service layer.
### How to test?
- You can't test this MR yet, we need to update some other parts.
- @Aristevao the tag [WIP] - work in progress means that this MR can't be merged when you finish the review.

### CODE REVIEW

File: src/api/services/PetService.ts [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
-    constructor() {
-        this.repository = new PetRepository();
+    constructor(connection: Connection) {
+        this.repository = connection.getCustomRepository(PetRepository);
     }
 
-    public insert(pet: Pet): any {
-        return this.repository.insert(pet);
+    public async insert(pet: Pet): Promise<Pet> {
+        return this.repository.save(pet);
```
Comment (Aristevao) [inline src/api/services/PetService.ts]: @bigLucas 
This `return` _(line 13)_ doesn't need to have an `await` ?

**Comment by Aristevao** [comment]
I read your changes. It's pretty similar to the one found in PF project that I took a look recently (_Acquirer_).

File: src/api/services/PetService.ts (hunk 1 above)
Comment (bigLucas) [reply src/api/services/PetService.ts]: @Aristevao 
Yes, I'll change it.

Comment (bigLucas) [reply src/api/services/PetService.ts]: @Aristevao 
Please **update** this branch, you can make a `commit` in this branch and make a `push` with the change.
Do you know how you can do this?

Comment (Aristevao) [reply src/api/services/PetService.ts]: @bigLucas 
Are the steps below?
```
git fetch --all
git checkout M-0008_refactoring_repository_layer_of_pet_entity
git pull
~ Make the changes
git add .
git commit -m "<message>"
git push
```

Comment (bigLucas) [reply src/api/services/PetService.ts]: Yes.
```shell
git push origin M-000...branch_name...
```

**Comment by bigLucas** [comment]
Hello @Aristevao, I made a new push to this PR.
Please see [here](https://github.com/Aristevao/serverlessoffline/pull/11/commits/1261eb3d0948cc0a43d6121d01754829289d2aa3)
**Note**: 
1. I made some fixes to use the `sqlite` database with the `webpack`.
2. We can't use the `sqlite` in production, on `aws` structure, because the database is a local database and at `aws` we don't have the access to local files (it's an AWS lambda behavior). This is not a problem, because this project is a sample to run offline, but if we want to run on `aws` we can't use this repository as a base.
3. You need to update the branch of  #16 PR.

File: src/api/core/Models.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { Pet } from "../../models/Pet";
+import { Tutor } from "../../models/Tutor";
+
+const __MODELS__ = [Pet, Tutor];
```
Comment (Aristevao) [inline src/api/core/Models.ts:4]: @bigLucas
In what context is the variable between underscore used?

**Comment by Aristevao** [comment]
> 3. You need to update the branch of  [[M-0012] Initialize database class and pass its connection to service layer #16](https://github.com/Aristevao/serverlessoffline/pull/16) PR.

By updating do you mean via `git merge`?

File: src/api/core/Models.ts (hunk 2 above)
Comment (bigLucas) [reply src/api/core/Models.ts:4]: I just copied this code from onother project... It is not a name convention, we could choose any name.

**Comment by bigLucas** [comment]
> > 3. You need to update the branch of  [[M-0012] Initialize database class and pass its connection to service layer #16](https://github.com/Aristevao/serverlessoffline/pull/16) PR.
> 
> By updating do you mean via `git merge`?
> 

Yes, the #16 MR is a merge of the M-0012 branch into the M-0008 branch. So, when we update the M-0008 branch we need to update the other branch (M-0012) with git merge.

**Comment by bigLucas** [comment]
@Aristevao this PR has a lot of changes that we already tested in other PRs, but it will be good if you make a final review, after the review you can let a thumbs up (:+1:) on the PR description and then we can make the merge.

**Comment by Aristevao** [comment]
> @Aristevao this PR has a lot of changes that we already tested in other PRs, but it will be good if you make a final review, after the review you can let a thumbs up (👍) on the PR description and then we can make the merge.

The final review is done!

**Pull request merged**